docs/core_docs/docs/integrations/document_loaders/web_loaders/couchbase.mdx

```diff
@@ -0,0 +1,104 @@
+---
+hide_table_of_contents: true
+sidebar_class_name: node-only
+---
+
+# Couchbase
+
+[Couchbase](http://couchbase.com/) is an award-winning distributed NoSQL cloud database that delivers unmatched versatility, performance, scalability, and financial value for all of your cloud, mobile, AI, and edge computing applications.
+
+This guide shows how to use load documents from couchbase database.
+
+# Installation
+
+```bash npm2yarn
+npm install couchbase
+```
+
+
+## Usage
+
+### Querying for Documents from Couchbase
+For more details on connecting to a Couchbase cluster, please check the [Node.js SDK documentation](https://docs.couchbase.com/nodejs-sdk/current/howtos/managing-connections.html#connection-strings).
+
+For help with querying for documents using SQL++ (SQL for JSON), please check the [documentation](https://docs.couchbase.com/server/current/n1ql/n1ql-language-reference/index.html).
+
+```typescript 
+import { CouchbaseDocumentLoader } from "langchain/document_loaders/web/couchbase";
+import { Cluster } from "couchbase"
+
+const connectionString = "couchbase://localhost";  // valid couchbase connection string
+const dbUsername = "Administrator"  // valid database user with read access to the bucket being queried
+const dbPassword = "Password"  // password for the database user
+
+// query is a valid SQL++ query
+const query = `
+        SELECT h.* FROM `travel-sample`.inventory.hotel h 
+        WHERE h.country = 'United States'
+        LIMIT 1
+    `;
+```
+
+### Connect to Couchbase Cluster
+
+```typescript
+  const couchbaseClient = await Cluster.connect(connectionString, {
```

Comment:
Formatting is a bit off here - you could also write this as a Deno notebook if you'd like?

https://github.com/langchain-ai/langchainjs/blob/main/docs/core_docs/docs/expression_language/streaming.ipynb

Reply:
I have fixed the formatting